pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -75,6 +76,32 @@ func NewMemServiceDiscovery(services map[model.Hostname]*model.Service, versions
 	}
 }
 
+// SyncStatus -- Not sure of the correct place to put this? Needed in istioctl and here.
+type SyncStatus struct {
+	ProxyID string `json:"proxy,omitempty"`
+	Sent    string `json:"sent,omitempty"`
+	Acked   string `json:"acked,omitempty"`
+}
+
+// Syncz dumps the synchronization status of all Envoys connected to this Pilot instance
+func Syncz(w http.ResponseWriter, req *http.Request) {
+	syncz := []SyncStatus{}
+	for _, con := range adsClients {
```

Comment:
Just curious, can istioctl request and parse adsz directly, instead of having this new handler?

Reply:
In order to enable upcoming diff-ing between proxy and Pilot adsz will need to move closer to the Envoy admin API config dump protos.

Reply:
Also adsz is already very large. I'm already skeptical about its usefulness with > 30 pods connected to Pilot.